clients/include/rocblas_arguments.hpp

```diff
@@ -324,4 +328,216 @@ static_assert(std::is_trivial<Arguments>{},
               "Arguments is not a trivial type, and thus is "
               "incompatible with C.");
 
+enum arg_type
+{
+    e_M,
+    e_N,
+    e_K,
+    e_lda,
+    e_ldb,
+    e_ldc,
+    e_ldd,
+    e_a_type,
+    e_b_type,
+    e_c_type,
+    e_d_type,
+    e_compute_type,
+    e_incx,
+    e_incy,
+    e_incb,
+    e_incd,
+    e_alpha,
+    e_beta,
+    e_transA,
+    e_transB,
+    e_side,
+    e_uplo,
+    e_diag,
+    e_batch_count,
+    e_stride_a,
+    e_stride_b,
+    e_stride_c,
+    e_stride_d,
+    e_stride_x,
+    e_stride_y,
+    e_algo,
+    e_solution_index,
+
+};
+
+#define CTOKEN(n) const char* const c_tok_##n = #n
+CTOKEN(M);
+CTOKEN(N);
+CTOKEN(K);
+CTOKEN(lda);
+CTOKEN(ldb);
+CTOKEN(ldc);
+CTOKEN(ldd);
+CTOKEN(a_type);
+CTOKEN(b_type);
+CTOKEN(c_type);
+CTOKEN(d_type);
+CTOKEN(compute_type);
+CTOKEN(incx);
+CTOKEN(incy);
+CTOKEN(incb);
+CTOKEN(incd);
+CTOKEN(alpha);
+CTOKEN(beta);
+CTOKEN(transA);
+CTOKEN(transB);
+CTOKEN(side);
+CTOKEN(uplo);
+CTOKEN(diag);
+CTOKEN(batch_count);
+CTOKEN(stride_a);
+CTOKEN(stride_b);
+CTOKEN(stride_c);
+CTOKEN(stride_d);
+CTOKEN(stride_x);
+CTOKEN(stride_y);
+CTOKEN(algo);
+CTOKEN(solution_index);
+#undef CTOKEN
+
+class ArgumentModel
```

Comment:
We should not need to define another class for a "16-argument model". Then we have two `Arguments` classes to maintain, instead of one universal class. Right now so much of the rocBLAS testing and benchmarking code, both C++ and Python, is wound up in the `Arguments` structure, that introducing a new class would not be good.

Better to add custom functions/methods which operate on the universal `Arguments` class.

Reply:
This "model" is a class for a different purpose, thus a different class.   I don't follow your argument we can discuss on Monday.

Reply:
This should not be used to create tools. They should be created outside of rocBLAS.

The code took 4 months to refactor to use `Arguments` and a YAML source model; this significantly deviates from that design.

Besides, it should not go into `rocblas_arguments.hpp`. I have already moved it out of that file, and am refactoring the logging.

Test generation, logging, and `Arguments` are 3 distinct issues, and should not be combined like this.

Reply:
There are already tools to dump YAML source which can be turned into tests.

A Python script can be written to take the YAML output and turn into a format compatible with  `*_gtest.yaml`.

This should be done by taking the raw YAML output from the logging, and transform it using Python code, rather than adding an `ArgumentsModel` to the C++ code.